sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -637,17 +765,53 @@ def upsert_entity(self, entity: EntityType, mode: UpdateMode = UpdateMode.MERGE,
                 :dedent: 16
                 :caption: Replacing/Merging or Inserting an entity into a table
         """
-        entity_json = _add_entity_properties(entity)
+
+    @overload
+    def upsert_entity(
+        self,
+        entity: T,
+        mode: UpdateMode = UpdateMode.MERGE,
+        *,
+        encoder: TableEntityEncoderABC[T],
+        **kwargs
+    ) -> Dict[str, Any]:
+        """Updates (merge or replace) an entity into a table.
+
+        :param entity: The properties for the table entity.
+        :type entity: Custom entity type
+        :param mode: Merge or Replace entity.
+        :type mode: ~azure.data.tables.UpdateMode
+        :keyword encoder: The encoder used to serialize the outgoing Tables entities.
+        :type encoder: ~azure.data.Tables.TableEntityEncoderABC
+        :return: Dictionary mapping operation metadata returned from the service.
+        :rtype: dict[str, Any]
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`
+        """
+    
+    @distributed_trace
+    def upsert_entity(self, *args, **kwargs) -> Dict[str, Any]:
+        entity = kwargs.pop("entity", None)
+        if not entity:
+            entity = args[0]
+        mode = kwargs.pop("mode", None)
+        if not mode:
+            try:
+                mode = args[1]
+            except IndexError:
+                mode = UpdateMode.MERGE
+        encoder = kwargs.pop("encoder", DEFAULT_ENCODER)
+        entity_json = encoder.encode_entity(entity)
         partition_key = entity_json["PartitionKey"]
         row_key = entity_json["RowKey"]
```

Comment:
Same comment - I think we should safely be able to pop these two values.

Reply:
Updated